src/components/PostDetails.tsx

```diff
@@ -2,117 +2,52 @@ import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
-import * as commentsApi from '../api/comments';
+import { useAppSelector, useAppDispatch } from '../app/hooks';
+import * as commentsActions from '../features/Slices/comments';
 
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
-
-type Props = {
-  post: Post;
-};
-
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+export const PostDetails: React.FC = () => {
+  const { comments, isLoading, error } = useAppSelector(
+    state => state.comments,
+  );
```

Comment:
It's a good practice to encapsulate the logic of selecting data from the state in selector functions. It makes the code more maintainable and easier to test.